### PR TITLE
Add alpha-dash-dot validation and lowerCase function

### PR DESCRIPTION
**Pull request by fedenaka**
I added a lowerCase function in case we need the functionality for other parameters in the future. Tell me if you like it and if you don't, i change that.

### CODE REVIEW

File: app/Validation/Rules/AlphaDashDotRule.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+namespace App\Validation\Rules;
+
+class AlphaDashDotRule extends AbstractRule implements ValidationRule
+{
+    protected $identifier = 'alpha_dash_dot';
+    protected $message = 'Field :field should contain firstname and lastname separated by an alpha_dash, dot or space';
+
+    public function validate($attribute, $username): bool
+    {
+        return preg_match('/^[a-zA-Z0-9]+([_\s\-\s.]?[a-zA-Z0-9])*$/', $username);
```
Comment (belohlavek) [inline app/Validation/Rules/AlphaDashDotRule.php]: https://github.com/illuminate/validation/blob/master/Validator.php#L1739-L1741

File: app/Validation/Rules/AlphaDashDotRule.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace App\Validation\Rules;
+
+class AlphaDashDotRule extends AbstractRule implements ValidationRule
+{
+    protected $identifier = 'alpha_dash_dot';
+    protected $message = 'Field :field should contain firstname and lastname separated by an alpha_dash, dot or space';
```
Comment (belohlavek) [inline app/Validation/Rules/AlphaDashDotRule.php]: https://github.com/laravel/laravel/blob/master/resources/lang/en/validation.php#L20

**Comment by belohlavek** [comment]
Check the changed files!

**Comment by fedenaka** [comment]
I have a question, should we let them set multiple dots/dashes/underscores or numbers only?
Ie: morty.mcloop.jr, 352352.

**Comment by belohlavek** [comment]
I think so, yep. As long as there are no leading/trailing symbols.
Leading/trailing numbers should be allowed tho.

**Pull request closed without merging**